src/component.js

```diff
@@ -196,6 +196,7 @@ export default class Component {
     }
 
     executeAndClearRemainingShowDirectiveStack() {
+        if (this.showDirectiveStack.length === 0) return false
```

Comment:
What's the `true/false` output supposed to represent? Whether there were any `show directives` to run?

In that case I would suggest creating a separate function to check that:
`hasAnyShowDirectivesToRun(){ return this.showDirectiveStack.length > 0 }`

Reply:
What about directly checking the `showDirectiveStack` length in `updateElements` something like
```
if (this.showDirectiveStack.length > 0) {
    this.executeAndClearRemainingShowDirectiveStack();
} else {
    this.executeAndClearNextTickStack(rootEl);
}
```